src/Sema.zig

```diff
@@ -1754,9 +1754,9 @@ fn analyzeBodyInner(
                 const err_union = try sema.resolveInst(extra.data.operand);
                 const err_union_ty = sema.typeOf(err_union);
                 if (err_union_ty.zigTypeTag(zcu) != .error_union) {
-                    return sema.fail(block, operand_src, "expected error union type, found '{}'", .{
-                        err_union_ty.fmt(pt),
-                    });
+                    const msg = try sema.errMsg(operand_src, "expected error union type, found '{}'", .{err_union_ty.fmt(pt)});
+                    try sema.errNote(operand_src, msg, "consider omitting 'try'", .{});
+                    return sema.failWithOwnedErrorMsg(block, msg);
```

Comment:
This will leak `msg` if `sema.errNote()` fails - if you look at other uses of `errNote()` there is a common pattern of using a labelled block with `errdefer msg.destroy(sema.gpa);` to prevent leaking the parent message. This issue is also present at the two other place where the new note it added.